server/src/main/java/io/crate/types/Regclass.java

```diff
@@ -57,9 +59,11 @@ public static Regclass primaryOid(RelationInfo relation) {
         );
     }
 
-    public static Regclass fromRelationName(String name) {
+    public static Regclass fromRelationName(String name, String currentSchema) {
+        name = trim(name, '"');
```

Comment:
Again not 100% related to the PR:
Postgres:
```
postgres=# select '""marios'""::regclass;
ERROR:  zero-length delimited identifier at or near """"
LINE 1: select '""marios'""::regclass;
```
Crate:
```
cr> select '""marios""'::regclass;                                                                                                                       
+-------------+
| -1459601621 |
+-------------+
| -1459601621 |
+-------------+
SELECT 1 row in set (0.001 sec)
```

Reply:
Probably a type and you meant:
```
postgres=# select '""marios""'::regclass;
ERROR:  invalid name syntax
LINE 1: select '""marios""'::regclass;
```
I guess PG is parsing the string literal as a `name` type if casted to `regclass`. 
PG also allows to use such identifiers as a table name:
```
postgres=# create table """t2""" (id int);
CREATE TABLE
postgres=# \dt
              List of relations
 Schema |      Name       | Type  |  Owner   
--------+-----------------+-------+----------
 public | "t2"            | table | postgres
(1 row)
postgres=# select '"""t2"""'::regclass;
 regclass 
----------
 """t2"""
(1 row)
```

CrateDB doesn't allow it:
```
cr> create table """t2""" (id int);                                                                                                                                                                                                                 
InvalidRelationName[Relation name ""t2"" is invalid.]
```

So I think it's fine that we simply trim double quotes. We could also use a parser to ensure that the string literal contains a valid identifier, but I doubt a bit that this is worth doing.

Reply:
I've pushed a commit to use the SQL parser on string to regclass casts.